### ISSUE TITLE
Indexing rate doesn't work for data streams

### ISSUE THREAD

**Issue by neeej** [acidvegas/elastop#4]
Indexing rate only seem to work if indices are written directly to them, not if they are data streams
![image](https://github.com/user-attachments/assets/c8cf0062-edf9-4d90-ad5e-bf320d702992)

I currently index around 5k/s to this cluster, using data streams.

And listing all indices in a huge list gets kind of unreadable when they have long names, like default APM indices have. They also get longer names set with "restored-" when using searchable snapshots, meaning even longer.

**Comment by acidvegas** [comment]
adding dynamic padding for the node/index names now.

**Comment by acidvegas** [comment]
> Indexing rate only seem to work if indices are written directly to them, not if they are data streams ![image](https://private-user-images.githubusercontent.com/35762408/390997415-c8cf0062-edf9-4d90-ad5e-bf320d702992.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4NjU1NTUsIm5iZiI6MTczMjg2NTI1NSwicGF0aCI6Ii8zNTc2MjQwOC8zOTA5OTc0MTUtYzhjZjAwNjItZWRmOS00ZDkwLWFkNWUtYmYzMjBkNzAyOTkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI5VDA3MjczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViZGJiODYxOWM2Mjg5ODU3ODIwMzQ0OTE5ZjI1MmRlNzcwYzQwNzVmMTIyNWI1MDk3MGU0YWU4YWI3ZjU5MTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nVNEX171BDlETkViwhyaEji3ETDRMbrmA8W3xoN-2KA)
> 
> I currently index around 5k/s to this cluster, using data streams.
> 
> And listing all indices in a huge list gets kind of unreadable when they have long names, like default APM indices have. They also get longer names set with "restored-" when using searchable snapshots, meaning even longer.

[4005778](https://github.com/acidvegas/elastop/commit/40057785124c4b72b98fe5388a63a3d9c8d45c02) should solve this issue. you can now hide some of the boxes. let me know if that works or if you have a better suggested idea

**Comment by neeej** [comment]
That looks better when side boxes are hidden. Thanks 👍 

Just missing the data streams info in the index list.

**Comment by acidvegas** [comment]
> That looks better when side boxes are hidden. Thanks 👍
> 
> Just missing the data streams info in the index list.

Awesome. Ill get those data stream rates fixed in the morning

**Comment by acidvegas** [comment]
> That looks better when side boxes are hidden. Thanks 👍
> 
> Just missing the data streams info in the index list.

Does [7d2187a](https://github.com/acidvegas/elastop/commit/7d2187ae85ca259179161811e0ef3a0db15607f8) show it how you would like? I made it so when you press the "h" key it toggles displaying hidden indexes (they show on the next refresh so give it a second)

Let me know if that it what you wanted



edit: I see, data streams are entirely different. I dont think I have used them myself personally. I will experiment with these tonight and maybe add a new box for data streams

**Comment by neeej** [comment]
> > That looks better when side boxes are hidden. Thanks 👍
> > Just missing the data streams info in the index list.
> 
> Does [7d2187a](https://github.com/acidvegas/elastop/commit/7d2187ae85ca259179161811e0ef3a0db15607f8) show it how you would like? I made it so when you press the "h" key it toggles displaying hidden indexes (they show on the next refresh so give it a second)
> 
> Let me know if that it what you wanted
> 
> edit: I see, data streams are entirely different. I dont think I have used them myself personally. I will experiment with these tonight and maybe add a new box for data streams

I can at least see index rate on them, via their backing indices, so that is a good improvement 👍 
But yes, showing them just as the data stream in the list (or a separate list somehow, one for indices and one for data streams could work as well.)

Would be good if I can choose how to sort them by name, size or indexing rate as well.
And if I can search would be another huge improvement, as onesome clusters I have quite a long list of indices.

Nicely done so far though 👍

**Comment by acidvegas** [comment]
> > > That looks better when side boxes are hidden. Thanks 👍
> > > Just missing the data streams info in the index list.
> > 
> > 
> > Does [7d2187a](https://github.com/acidvegas/elastop/commit/7d2187ae85ca259179161811e0ef3a0db15607f8) show it how you would like? I made it so when you press the "h" key it toggles displaying hidden indexes (they show on the next refresh so give it a second)
> > Let me know if that it what you wanted
> > edit: I see, data streams are entirely different. I dont think I have used them myself personally. I will experiment with these tonight and maybe add a new box for data streams
> 
> I can at least see index rate on them, via their backing indices, so that is a good improvement 👍 But yes, showing them just as the data stream in the list (or a separate list somehow, one for indices and one for data streams could work as well.)
> 
> Would be good if I can choose how to sort them by name, size or indexing rate as well. And if I can search would be another huge improvement, as onesome clusters I have quite a long list of indices.
> 
> Nicely done so far though 👍

Can you see if [1caff77](https://github.com/acidvegas/elastop/commit/1caff777348ff296bcc800e8c2424b14deb23c9e) displays the data streams correctly for you? Let me know if anything shows out of alignment aswell. Should work fine. It will display a little icon indicator showing to show its a data stream.

You think it would be better to have a separate box for data streams or to keep them in the index list like I did in this commit?

**Comment by neeej** [comment]
> Can you see if [1caff77](https://github.com/acidvegas/elastop/commit/1caff777348ff296bcc800e8c2424b14deb23c9e) displays the data streams correctly for you? Let me know if anything shows out of alignment aswell. Should work fine. It will display a little icon indicator showing to show its a data stream.
> 
> You think it would be better to have a separate box for data streams or to keep them in the index list like I did in this commit?

Does not seem to work, looks like before, no data streams are listed.

The alignment works unless I remove box 3, then box 5 gets the same width as box 4, which then gets wrapped lines instead (depending on how large screen resolution I have it on).

As it's a "top" app, I would say I would mostly benefit from knowing what indices are currently being written to (sorted by that).
Meaning I wouldn't mind having it as is in a single list for just the indices.
Maybe if the list could be shortened with an option to expand it, as with lots of indices in the cluster (I have 1400 in one of my clusters) elastop seem to be a bit too slow.

Or maybe data streams aren't needed to be shown separate at all, just that the backing indices somehow are shown that they are datastreams might work as well.

**Comment by acidvegas** [comment]
> > Can you see if [1caff77](https://github.com/acidvegas/elastop/commit/1caff777348ff296bcc800e8c2424b14deb23c9e) displays the data streams correctly for you? Let me know if anything shows out of alignment aswell. Should work fine. It will display a little icon indicator showing to show its a data stream.
> > You think it would be better to have a separate box for data streams or to keep them in the index list like I did in this commit?
> 
> Does not seem to work, looks like before, no data streams are listed.
> 
> The alignment works unless I remove box 3, then box 5 gets the same width as box 4, which then gets wrapped lines instead (depending on how large screen resolution I have it on).
> 
> As it's a "top" app, I would say I would mostly benefit from knowing what indices are currently being written to (sorted by that). Meaning I wouldn't mind having it as is in a single list for just the indices. Maybe if the list could be shortened with an option to expand it, as with lots of indices in the cluster (I have 1400 in one of my clusters) elastop seem to be a bit too slow.
> 
> Or maybe data streams aren't needed to be shown separate at all, just that the backing indices somehow are shown that they are datastreams might work as well.

Yeah, I didn't even think about that...while I do have about 2tb of data currently, I am only working with like 10-20 indexes at the moment.

I am going to see if I can maybe stack requests in an API call or maybe specify only the params needed from the API endpoint to truncate the response.

As far as sorting the indexes go, [1d08518](https://github.com/acidvegas/elastop/commit/1d085188473bd2b015aa5ee6a4c0199c374a3e0f) now sorts actively written to indexes to the top.


Also:
>  Meaning I wouldn't mind having it as is in a single list for just the indices.

Does it not work when you hide all the boxes and only show the indices box?


Either way, I do need to make this more dynamic. Maybe a yaml config in $HOME/.config to specify the default startup boxes, columns to display, etc

Also, if your console shows anything not lining up or displaying properly, please share pictures if you can so I can fix it.

Appreciate all the feed back btw

**Comment by neeej** [comment]
Looks much better now when it sorts on index rate 👍 

I did noticed the nodes are line break'd as well now, unless I run this in full screen on a high resolution, I guess the Roles column could be a bit smaller.
And this is with box 3 hidden:
![image](https://github.com/user-attachments/assets/5a42e3ec-a392-4b68-9c64-e1d7ebd99021)

So, it looks fine for the index list unless only box 4 and 5 are shown, then they get the same amount of width, and indices are then line breaked, as per above image.

Not sure what some of those cluster metrics (in box 5) calculate either, as it looks like the "Rate" one's are a bit off.

One other thing I'd love to see would be how many shards a node has on it, that would be super helpful.
